File: src/backend/gemini_api.py
import google.generativeai as genai
import os
from dotenv import load_dotenv


def call_gemini(input: str) -> str:
    """
    for now this is just a function just to get see how gemini runs
    if we need to we make make the api return like a json if we need
    it too
    """

    load_dotenv()
    genai.configure(api_key = os.getenv("API_KEY"))
    model = genai.GenerativeModel("gemini-1.5-flash")
    prompt = f"""Context: you are a doctor, and your job is to diagnose me 
    with a few possible illness i may have given the symptoms: {input}. When sending 
    your response back, only return a list of possible illness, no questions and no other 
    filler text
    """

    
    response = model.generate_content(prompt)
    #response includes other information outside of its text if we need it
    return response.text


"""

print("[OUTPUT OF GEMINI]")
print(call_gemini("headache, fever, runny rose, sore throat"), end="")
print("[END OUTPUT]")

"""

File: src/backend/gemini_api.js
import dotenv from 'dotenv/config';


import { GoogleGenerativeAI } from "@google/generative-ai";

async function call_gemini(input) {
    const genAI = new GoogleGenerativeAI(process.env.API_KEY);
    const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });

    const prompt = `Context: you are a doctor, and your job is to diagnose me 
    with a few possible illness i may have given the symptoms: ${input}. When sending 
    your response back, only return a list of possible illness, no questions and no other 
    filler text`


    const result = await model.generateContent(prompt);
    console.log(result.response.text());

    return result.response.text()
}


//const test = call_gemini("fever, cough, shortness of breath");